src/GslCore/ResolveExtPart.fs

```diff
@@ -2,208 +2,122 @@
+    let part = fetchPart pid |> returnOrFail
 
-                // Look for simple case.  If we are just using the part from the hutch unadulterated, then
-                // we specify things differently, referring to the external id
-                if partId.mods.Length = 0 then
-                    let dna = if ppp.fwd then dna else dna.RevComp()
-                    {id = None; 
-                     extId = Some(pid.[1..]); 
-                     sliceName = sliceName;
-                     uri = uri; // TODO: use the URI of rabit from hutch here instead?
-                     dna = dna; 
-                     sourceChr = "library";
-                     sourceFr = 0<ZeroOffset>;
-                     sourceTo = (dna.Length-1)*1<ZeroOffset>;
-                     sourceFwd = ppp.fwd;
-                     sourceFrApprox = false;
-                     sourceToApprox = false;
-                     // Don't assign coordinates to pieces until later when we decide
-                     // how they are getting joined up
-                     destFr = 0<ZeroOffset>; 
-                     destTo = 0<ZeroOffset>; 
-                     destFwd = ppp.fwd;
-                     description = rabit.Name; 
-                     sliceType = REGULAR; 
-                     amplified = false;
-                     template = Some dna; // not amplifying from this
-                     dnaSource =
-                        match ppp.pr.TryGetOne("dnasrc") with
-                        | Some(d) -> d
-                        | None -> pid;
-                     pragmas = ppp.pr;
-                     breed = B_X; // will be replaced at final submission
-                     materializedFrom = Some(ppp);
-                     annotations = []} // FIXME: need to generate annotations based on Rabit metadata
-                else
-                    // Otherwise, they are taking a hutch part and doing something to it,
-                    // so the hutch is just another DNA source and they are effectively
-                    // building a new rabit
+    // Have part from the hutch.  We might just use it verbatim or we might be
+    // some modifications to it to make a new part
+    let part = fetchPart pid |> returnOrFail
 
-                    // Start off assuming it's the full DNA slice
-                    let startSlice =
-                       {left = {x = 1<OneOffset>; relTo = FivePrime};
-                        lApprox = false; 
-                        rApprox = false;
-                        right = {x = -1<OneOffset>; relTo = ThreePrime}}
+    // Check for slice modifications.  We can't handle any other type of mod at this point, so
+    // ensure there are none.
+    if partId.mods |> List.exists (fun m -> match m with | SLICE(_)-> false | _ -> true) then
+        failwithf "ERROR:  could not process mods for rabit %s %A\n" partId.id partId.mods

```

Comment:
Possibly use a more generic term than 'rabit' for the error message here

Reply:
I didn't actually change this, it probably just moved around.